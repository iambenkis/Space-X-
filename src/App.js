import React, { useEffect } from 'react';
import {
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Rocket from './pages/Rocket/Rocket';
import Mission from './pages/Mission/Mission';
import Profile from './pages/Profile/Profile';
import Header from './components/Header/Header';
import { getRockets } from './redux/Rockets/Rockets/rockets';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getRockets());
  }, [dispatch]);
  return (
    <>
      <Header />
      <Routes>
        <Route path="/rockets" element={<Rocket />} />
        <Route path="/" element={<Navigate replace to="/rockets" />} />
        <Route path="/missions" element={<Mission />} />
        <Route path="/profile" element={<Profile />} />
      </Routes>
    </>
  );
};

export default App;
