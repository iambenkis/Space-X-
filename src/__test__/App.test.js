import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/Rockets/configurationStore';
import App from '../App';

test('App render correctly', () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </BrowserRouter>,
  );
});

test('App snapshot testing', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </BrowserRouter>,
  );
  expect(tree).toMatchSnapshot();
});
