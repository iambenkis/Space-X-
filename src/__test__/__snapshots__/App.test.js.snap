// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot testing 1`] = `
Array [
  <header
    className="header"
  >
    <a
      className="logo"
      href="/"
    >
      <img
        alt="Space Traveler's Hub"
        className="logo-image"
        src="planet.png"
      />
      Space Traveler's Hub
    </a>
    <nav
      className="nav"
    >
      <ul
        className="unorderList"
      >
        <li
          className="list"
        >
          <a
            aria-current="page"
            className="link active"
            href="/rockets"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li
          className="list"
        >
          <a
            className="link "
            href="/missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        <li
          className="list"
        >
          <a
            className="link "
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="rockets"
  />,
]
`;
